include/net/net_if.h

```diff
@@ -1600,6 +1600,14 @@ static inline bool net_if_is_up(struct net_if *iface)
  */
 int net_if_down(struct net_if *iface);
 
+/**
+ * @brief Tell network interface that carrier went down. This is typically
+ * called by network device driver when network connection is lost.
+ *
+ * @param iface Pointer to network interface
+ */
+void net_if_carrier_down(struct net_if *iface);
```

Comment:
huh no, carrier is really something ethernet centric. Actually see net/ethernet_mgmt.h there are ethernet_mgmt_raise_carrier_<on/off>_event() functions

that said, for a more generic stuff we could have a link_up and link_down as status flag.

However, that requires a bit of thought to get this right. We had that discussion already back in the days with @Vudentz when he introduced up/down status flag and getting all these states proper would need some changes. up/down would be about IP core being ready/off on the iface. link up/down would be about the hw being ready (for instance, in case of ethernet there would no reason to put the link up if the carrier is off).  etc...

Reply:
Hmm, net_if_carrier_down() is "generic" and already used by Bluetooth (see subsys/net/ip/l2/bluetooth.c)

Reply:
that makes sense because before being able to do ipsp, you need an actual bt connection below.
Like in ethernet you need an actual cable being plugged in (carrier line).

So carrier is still specific to certain bearers then. There is no such things in 15.4 for instance or slip tap etc...

I can't recall exactly what was the discussion, it was on gerrit back then (if I remember well) so that's a long time ago. Idea was the we need to settle different status flags for exact proper state of device/iface so it's not messed up through device drivers, l2 and ip core stack.

Well, we probably don't have time for 1.12 to think that through so let's make it easier, it will be already an improvement: add net_if_carrier_up(), the Ethernet device driver would call that one. In net_if.c then call net_if_up() there. This will enable future features like not automatically "connect" (putting the iface up with an ip etc...) the device.